cli/azd/internal/vsrpc/utils.go

```diff
@@ -0,0 +1,45 @@
+package vsrpc
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/apphost"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/azure/azure-dev/cli/azd/pkg/tools/dotnet"
+)
+
+// appHostServiceForProject returns the ServiceConfig of the service for the AppHost project for the given azd project.
+func appHostForProject(
+	ctx context.Context, pc *project.ProjectConfig, dotnetCli dotnet.DotNetCli,
+) (*project.ServiceConfig, error) {
+	for _, service := range pc.Services {
+		if service.Language == project.ServiceLanguageDotNet {
+			isAppHost, err := dotnetCli.GetMsBuildProperty(ctx, service.Path(), "IsAspireHost")
+			if err != nil {
+				log.Printf("error checking if %s is an app host project: %v", service.Path(), err)
```

Comment:
How does a user fetch debugging logs while interfacing with azd through VS in case something is awry?

Reply:
VS is running us with `--debug` and capturing `stderr` to report when they see failures, and I think they present that to users (they had captured some logs for me when I was running into concurrent map mutation and the runtime was panicking).

Long term we can think about leveraging the same `IObserver` technology to have just another stream of debug messages that VS can listen to and record and present to the user instead of capturing all standard error output.

Reply:
I missed how `--debug` ends up working as expected nicely without any additional changes. Just a single log file seems sufficient for the near to medium term. Pretty cool!